### ISSUE TITLE
Elasticsearch filter specs don't seem to support multiple instances properly

### ISSUE THREAD

**Issue by colinodell** [api-platform/core#5574]
**API Platform version(s) affected**: 2.4.0 and higher

## Description

**TL;DR: The OpenApi specs for Elasticsearch filters don't seem to denote array support properly.**

The Elasticsearch `AbstractSearchFilter` class defines filter parameters as both `foo` and `foo[]`:

https://github.com/api-platform/core/blob/4e1b0c92271a92da3dc4d285b560714f2d504c5f/src/Elasticsearch/Filter/AbstractSearchFilter.php#L94-L101

Presumably this is done to denote that this filter can be used one or more times via query strings like:

- `?foo=aaa`
- `?foo[]=aaa&foo[]=bbb`

The generated OpenAPI schema currently looks like this:

```json5
"parameters": [
    {
        "name": "foo",
        "in": "query",
        "description": "",
        "required": false,
        "deprecated": false,
        "allowEmptyValue": true,
        "schema": {
            "type": "string"
        },
        "style": "form",
        "explode": false,
        "allowReserved": false
    },
    {
        "name": "foo[]",
        "in": "query",
        "description": "",
        "required": false,
        "deprecated": false,
        "allowEmptyValue": true,
        "schema": {
            "type": "string"
        },
        "style": "form",
        "explode": false,
        "allowReserved": false
    },
]
```

However, because the `schema` for `foo[]` says it's a `string`, some validation tools (like [`league/openapi-psr7-validator`](https://packagist.org/packages/league/openapi-psr7-validator)) will complain about `?foo[]=1&foo[]=2` saying that `Value expected to be 'string', but 'array' given`.

Additionally, Swagger UI will show both `foo` and `foo[]` as only allowing a single `string` value:

<img width="555" alt="image" src="https://user-images.githubusercontent.com/202034/234982350-a53d72d7-cd1c-4a86-86ae-3092f46005c4.png">

I suspect that this behavior is probably not what we want.

## How to reproduce

Add an Elasticsearch `MatchFilter` or `TermFilter` to any API resource and then:

 - Inspect the generated spec
 - Look at the filters in the SwaggerUI
 - Validate some requests that use the same filter multiple times

## Possible Solution

I think we could replace the code shown earlier with this:

```php
$description[$property] = [
    'property' => $property,
    'type' => $hasAssociation ? 'string' : $this->getPhpType($type),
    'required' => false,
];

$description["${property}[]"] = [
    'property' => $property,
    'type' => 'array',
    'schema' => [
        'type' => 'array',
        'items' => [
            'type' => $hasAssociation ? 'string' : $this->getPhpType($type),
        ],
    ],
    'required' => false,
];
```

This fixes the SwaggerUI:

<img width="553" alt="image" src="https://user-images.githubusercontent.com/202034/234983523-cabda922-15cc-453b-837a-0a748c0592bd.png">

And results in URLs like:

- `/api/some-resource?foo%5B%5D=aaa&foo%5B%5D=bbb`

This approach also aligns with how other developers have defined specs to support PHP's special array syntax in query params:

 - https://stackoverflow.com/a/68459831
 - https://stackoverflow.com/a/43465801 

If you feel this suggestion is correct and reasonable, I'd be happy to submit a PR - just let me know which branch to use as my base :)

**Comment by colinodell** [comment]
Bumping this, as I think the proposed solution could improve DX if the maintainers feel that it's a valid approach.
